hack/generator/pkg/astmodel/type_visitor.go

```diff
@@ -28,7 +28,7 @@ type TypeVisitor struct {
 	VisitResourceType  func(this *TypeVisitor, it *ResourceType, ctx interface{}) (Type, error)
 	VisitFlaggedType   func(this *TypeVisitor, it *FlaggedType, ctx interface{}) (Type, error)
 	VisitValidatedType func(this *TypeVisitor, it ValidatedType, ctx interface{}) (Type, error)
```

Comment:
minor: Wondering if we want `ValidatedType` to be a ptr since basically everything else is?

Reply:
Yes it should be, as it will blow up in the same way with `==`. Not sure how I missed it.